component/src/main/java/org/wso2/extension/siddhi/io/cdc/source/CDCSource.java

```diff
@@ -419,10 +420,22 @@ public void connect(ConnectionCallback connectionCallback) throws ConnectionUnav
 
     @Override
     public void disconnect() {
+        if (mode.equals(CDCSourceConstants.MODE_POLLING)) {
+            HikariDataSource dataSource = cdcPoller.getDataSource();
+            boolean isLocalDatasource = cdcPoller.isLocalDataSource();
+            if (dataSource != null && isLocalDatasource) {
+                dataSource.close();
+                if (log.isDebugEnabled()) {
+                    log.debug("Closing the pool name: " + dataSource.getPoolName());
+                }
+            }
+        }
     }
 
     @Override
     public void destroy() {
+        this.disconnect();
```

Comment:
After disconnecting before the shutdown, is there a possibility of a poll?

Reply:
Fixed with commit:  23c9f82